docs/source/user_guide/intro.rst

```diff
@@ -1,107 +1,24 @@
 Introduction to DoWhy
 =====================
+*[Maybe also include: "Who is this library/user guide for?"]*
 
-The need for causal inference
-----------------------------------
+DoWhy helps to perform a wide variety of *causal tasks* based on a method generally referred to as causal inference.
+These tasks help to give answers to questions, such as "If I change the color of my button to red, how much will this change users’ purchase decisions (effect)?", or "Which service in my distributed system has caused the frontend to be slower than usual?".
 
-Predictive models uncover patterns that connect the inputs and outcome in observed data. To intervene, however, we need to estimate the effect of changing an input from its current value, for which no data exists. Such questions, involving estimating a *counterfactual*, are common in decision-making scenarios.
+To give readers easy access to learn about such tasks, this user guide dedicates the entire chapter
+:doc:`causal_tasks/index` to explain how DoWhy can be used to perform
+such tasks. We categorize tasks into :doc:`causal_tasks/estimating_causal_effects/index`,
+:doc:`causal_tasks/root_causing_and_explaining/index`, and :doc:`causal_tasks/what_if/index`.
 
-* Will it work?
-    * Does a proposed change to a system improve people's outcomes?
-* Why did it work?
-    * What led to a change in a system's outcome?
-* What should we do?
-    * What changes to a system are likely to improve outcomes for people?
-* What are the overall effects?
-    * How does the system interact with human behavior?
-    * What is the effect of a system's recommendations on people's activity?
+To perform tasks, DoWhy leverages two powerful frameworks, namely graphical causal models (GCM) and potential
+outcomes (PO). Depending on the task at hand, it leverages GCM, or PO, or both, or none. What’s common to all tasks
+though, except for unit-change and feature attribution, is that they require a *causal graph*, which is modeled after
+the problem domain. For that reason, this user guide starts with :doc:`modeling_causal_relations/index`. Readers
+interested in :doc:`causal_tasks/root_causing_and_explaining/unit_change`
+or :doc:`causal_tasks/root_causing_and_explaining/feature_attribution` can skip right to those sections.
 
```

Comment:
> Can maybe more generally say “Quantifying causal contributions” or something along the line instead of specifically ICC, seeing that we also have direct arrow strength.

Yes. The PowerPoint presentation (which acts as source for this png) is alongside everything else. Just open it, make the necessary changes and take a screenshot of the final result (sorry, PP's export features are a bit limited, but I like the easy editing of PP). Make the screenshot rather too big than too small to make sure it looks crisp when rendered in the docs.

> Would it make sense to add an optional validation/refutation step? Although this might be difficult seeing that this happens after the estimation step in case of effect estimation.

Yes, I think that makes sense. Same is true for the validation of GCMs. The fact that this is a b it hard to model visually, might also be an indicator that not everything is perfect API-wise, or documentation-wise, or about how we think about this stuff. So it's definitely open for iteration. OTOH, I wouldn't over-engineer the graphic, because it would lose its simplicity for navigation. I'm sure we can find a middleground.